packages/cli/src/commands/codegen/generate.ts

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020-2023 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import path from 'path';
+import {Command, Flags} from '@oclif/core';
+import {getProjectRootAndManifest} from '@subql/common';
+import * as inquirer from 'inquirer';
+import {
+  generateHandlers,
+  generateManifest,
+  getAbiInterface,
+  getAvailableEvents,
+  getAvailableFunctions,
+} from '../../controller/scaffoldgen-controller';
+
+export interface SelectedMethod {
+  name: string;
+  method: string;
+}
+export interface UserInput {
+  startBlock: number;
+  functions: SelectedMethod[];
+  events: SelectedMethod[];
+  abiPath: string;
+  address?: string;
+}
+
+export default class Generate extends Command {
+  static description = 'Create Project on Hosted Service';
+
+  static flags = {
+    location: Flags.string({
+      char: 'l',
+      description: '[deprecated] local folder to run codegen in. please use file flag instead',
+    }),
+    file: Flags.string({char: 'f', description: 'specify manifest file path (will overwrite -l if both used)'}),
+    events: Flags.string({description: 'abi events', required: true}),
+    functions: Flags.string({description: 'abi functions', required: true}),
+    abiPath: Flags.string({description: 'path to abi from root', required: true}),
+    startBlock: Flags.integer({description: 'startBlock', required: true}),
+    address: Flags.string({description: 'contract address'}),
+  };
+
+  async run(): Promise<void> {
+    const {flags} = await this.parse(Generate);
+    const {abiPath, address, events, file, functions, location, startBlock} = flags;
+    const projectPath = path.resolve(file ?? location ?? process.cwd());
+    const {manifests, root} = getProjectRootAndManifest(projectPath);
+
+    const abiInterface = getAbiInterface(projectPath, abiPath);
+
+    const eventsFragments = getAvailableEvents(abiInterface);
+    const functionFragments = getAvailableFunctions(abiInterface);
+
+    const availableEventList = Object.keys(eventsFragments);
+    const availableFunctionList = Object.keys(functionFragments);
+
+    const eventArray: string[] = [];
+    const functionArray: string[] = [];
+
+    if (events !== '*') {
+      try {
+        const chosenEvent = await inquirer.prompt({
+          name: 'events',
+          message: 'Select events',
+          type: 'checkbox',
+          choices: availableEventList,
+        });
+        eventArray.push(...chosenEvent.events);
+      } catch (e) {
+        throw new Error(e);
+      }
```

Comment:
This is redundant, if a catch just rethrows the error its the same as not having a try/catch